### DISCUSSION TITLE
PERMANOVA in adonis2: correct use of factors

### DISCUSSION THREAD

**Discussion by AylenM** [vegandevs/vegan#676] [Q&A]
Hi!
I'm really lost because I have read very different things of the same topic. 
I have a continous variable of interest (X), but also factors that I think they can affect my results (for me the most interesting factor is Sampling site). So, I have made this formula for adonis2:
adonis2(X ~ SamplingSite + Year + Sex + Age + SampleType, data=mydata,method = "euclidean", permutations = 99999)
(Sampling site has 2 levels, Year has 4 categories, Sex has 2, Age has 2, Sample type has 2)

-Are these factors "fixed" in this way?
-Is this formula right? I know that changing order makes results to be different. Maybe because not all factors have exactly the same n by category? My total n is also <30

Any help/advice/suggestions would be greatly appreciated!!
Thank you in advance!

**Comment by gavinsimpson** [comment]
> Are these factors "fixed" in this way?

What do you mean by "fixed"? If you mean are these factors fixed effects (as opposed to random effects) then yes, they are fixed effects.

> Is this formula right?

How would we know :-) You haven't told us what you are trying to do, what hypotheses you want to test etc.

> I know that changing order makes results to be different.

This is **not** true; it wasn't even true (AFAICR) in the old `adonis()`. Where this might come from is that with old `adonis()` we *only* had am omnibus test and a sequential test of terms. It was the latter test where order mattered. You can still do those sequential tests in `adonis2()`, but we now have `by = "margin"` which doesn't suffer the same problem.
